### PR TITLE
🐛 fix input text issue

### PR DESCRIPTION
**Pull request by prashantgupta24**
## Description
To fix the recent change in vllm - since output contains the input text they do not pass in the prompt with the result anymore.

## How Has This Been Tested?
Tested by locally copying code over to the dev_pod and running a request

### Single request
```
grpcurl -plaintext -proto proto/generation.proto -d \
  '{
    "model_id": "dummy-model-name",
    "requests": [
      {
        "text": "Once upon a time,"
      }
    ],
    "params": {
      "method": "GREEDY",
      "stopping": {
        "max_new_tokens": 20
      },
      "response": {
          "input_text": true
        }
    }
  }' \
  localhost:8033 fmaas.GenerationService/Generate
{
  "responses": [
    {
      "generatedTokenCount": 20,
      "text": "Once upon a time, there was a little girl who loved to read. She loved to read so much that she would read",
      "inputTokenCount": 6,
      "stopReason": "MAX_TOKENS"
    }
  ]
}
```
### 2 requests
```
❯ grpcurl -plaintext -proto proto/generation.proto -d \
  '{
    "model_id": "dummy-model-name",
    "requests": [
      {
        "text": "Once upon a time,"
      },
      {
        "text": "When I was little,"
      }
    ],
    "params": {
      "method": "GREEDY",
      "stopping": {
        "max_new_tokens": 20
      },
      "response": {
          "input_text": true
        }
    }
  }' \
  localhost:8033 fmaas.GenerationService/Generate
{
  "responses": [
    {
      "generatedTokenCount": 20,
      "text": "Once upon a time, there was a little girl who loved to read. She loved to read so much that she would read",
      "inputTokenCount": 6,
      "stopReason": "MAX_TOKENS"
    },
    {
      "generatedTokenCount": 20,
      "text": "When I was little, I was a big fan of the movie, The Wizard of Oz. I loved the movie",
      "inputTokenCount": 6,
      "stopReason": "MAX_TOKENS"
    }
  ]
}
```
### Streaming
```
❯ grpcurl -plaintext -proto proto/generation.proto -d \
  '{
    "model_id": "dummy-model-name",
    "request": [
      {
        "text": "Once upon a time,"
      }
    ],
    "params": {
      "method": "GREEDY",
      "stopping": {
        "max_new_tokens": 4
      },
      "response": {
          "input_text": true
        }
    }
  }' \
  localhost:8033 fmaas.GenerationService/GenerateStream
{
  "text": "Once upon a time,",
  "inputTokenCount": 6
}
{
  "generatedTokenCount": 1,
  "text": " there"
}
{
  "generatedTokenCount": 2,
  "text": " was"
}
{
  "generatedTokenCount": 3,
  "text": " a"
}
{
  "generatedTokenCount": 4,
  "text": " little",
  "stopReason": "MAX_TOKENS"
}
```

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: src/vllm_tgis_adapter/grpc/grpc_server.py [hunk 1]
```diff
@@ -461,7 +461,12 @@ def _convert_input_details(
 
         if resp_options.input_text:
             response.text = (
-                result.prompt if not response.text else result.prompt + response.text
+                result.prompt
+                if not response.text
+                else tokenizer.convert_tokens_to_string(
+                    tokenizer.convert_ids_to_tokens(result.prompt_token_ids)
```
Comment (prashantgupta24) [inline src/vllm_tgis_adapter/grpc/grpc_server.py:467]: not sure if there's a single method that can do both the conversions?

File: src/vllm_tgis_adapter/grpc/grpc_server.py [hunk 2]
```diff
@@ -461,7 +461,12 @@ def _convert_input_details(
 
         if resp_options.input_text:
             response.text = (
-                result.prompt if not response.text else result.prompt + response.text
+                result.prompt
+                if not response.text
```
Comment (maxdebayser) [inline src/vllm_tgis_adapter/grpc/grpc_server.py:465]: So if  `response.text` is None, we have the prompt text in `result.prompt`, but if it is not None, the `result.prompt` is discarded or never set?

Comment (prashantgupta24) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:465]: Hmm, good point. How can I test out the `response.text` as None?

**Review by maxdebayser — APPROVED** [review]
LGTM

**Comment by tjohnson31415** [comment]
I tried out this change out and noticed that the BOS token is included in the text output:
```
{
  "responses": [
    {
      "generatedTokenCount": 20,
      "text": "\u003cs\u003eleys, to flare upon the\ndarkness of their condition.  But, the time was not come yet; and\nevery wind that blew over France shook the rags of the scarecrows\nin vain, for the birds, fine of song and feather, took no warning.\n\nThe wine-shop was a corner shop, better than most others in its\nappearance and degage, but less comfortable inside.  It was, for the\nneighbourhood, a",
      "inputTokenCount": 85,
      "stopReason": "MAX_TOKENS"
    }
  ]
}
```

**Comment by prashantgupta24** [comment]
@tjohnson31415 thanks a lot for the suggestion to add the prompt directly instead of going down the complicated route of decoding the tokens!

File: src/vllm_tgis_adapter/grpc/grpc_server.py [hunk 3]
```diff
@@ -283,6 +287,8 @@ async def Generate(
             #     # Abort the request if the client disconnects.
             #     await self.engine.abort(f"{request_id}-{i}")
             #     return self.create_error_response("Client disconnected")
+            if res.prompt is None:
+                res.prompt = request_prompt_map[res.request_id]
```
Comment (tjohnson31415) [inline src/vllm_tgis_adapter/grpc/grpc_server.py:291]: I think we can remove the `request_prompt_map` and just use the index of the request as our id:
```suggestion
                res.prompt = request.requests[i].prompt
```

Comment (njhill) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:291]: can it just be list rather than dict?

Comment (tjohnson31415) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:291]: Actually, I do think we want the additional list/dict here, but it should contain the truncated prompt.

Comment (tjohnson31415) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:291]: Or maybe not... using the `/completions` endpoint with vLLM with `echo: "true"` seems to also return the un-truncated text in its output.

Comment (prashantgupta24) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:291]: @tjohnson31415 I thought about doing like that, are we 100% sure that the order of the requests and the responses will line up each time? The map made sure we got the exact text based on the response

File: src/vllm_tgis_adapter/grpc/grpc_server.py [hunk 4]
```diff
@@ -260,11 +260,12 @@ async def Generate(
                 kwargs["trace_headers"] = extract_trace_headers(headers)
             elif contains_trace_headers(headers):
                 log_tracing_disabled_warning()
+            unique_request_id = f"{request_id}-{i}"
             generators.append(
                 self.engine.generate(
                     inputs=inputs,
                     sampling_params=sampling_params,
-                    request_id=f"{request_id}-{i}",
+                    request_id=unique_request_id,
```
Comment (tjohnson31415) [inline src/vllm_tgis_adapter/grpc/grpc_server.py:268]: Changs LGTM. One more NIT and then also squash the commits for the rebase merge.

NIT: We can remove the extra `unique_request_id` variable now after the other changes.

**Review by tjohnson31415 — APPROVED** [review]
LGTM!

**Pull request merged**